core/domain/collection_services.py

```diff
@@ -1243,34 +1244,17 @@ def get_next_page_of_all_non_private_commits(
     ) for entry in results], new_urlsafe_start_cursor, more)
 
 
-def _should_index(collection):
-    """Checks if a particular collection should be indexed.
+def _collection_rights_to_search_dict(rights):
+    """Returns a search dict with information about the collection rights. This
+    allows searches like "is:featured".
 
     Args:
-        collection: Collection.
+        rights: ActivityRights. Rights object for a collection.
     """
-    rights = rights_manager.get_collection_rights(collection.id)
-    return rights.status != rights_manager.ACTIVITY_STATUS_PRIVATE
-
-
-def _collection_to_search_dict(collection):
-    """Converts a collection domain object to a search dict.
-
-    Args:
-        collection: Collection. The collection domain object to be converted.
 
-    Returns:
-        The search dict of the collection domain object.
-    """
-    doc = {
-        'id': collection.id,
-        'title': collection.title,
-        'category': collection.category,
-        'objective': collection.objective,
-        'language_code': collection.language_code,
-        'tags': collection.tags,
-        'rank': _DEFAULT_RANK,
-    }
+    doc = {}
+    if rights.status == rights_manager.ACTIVITY_STATUS_PUBLICIZED:
```

Comment:
Hm, did you merge correctly? I thought we removed ACTIVITY_STATUS_PUBLICIZED altogether from the codebase. (Please take a look at develop to see what's currently there and what isn't, and update the PR accordingly.)